src/lib/UtapiReplay.js

```diff
@@ -187,11 +186,14 @@ export default class UtapiReplay {
         }
         const replay = scheduleJob(this.replaySchedule, () => {
             // Ensure no elements are still being pushed by a prior replay.
-            if (!this.replayLock) {
-                this._setReplayLock(true);
-                return this._checkLocalCache();
-            }
-            return undefined;
+            this.localCache.setExpire('s3:utapireplay:lock', 'true', TTL)
```

Comment:
I liked the original approach of using an abstract method for locking `_setReplayLock()`